javaslang/src/main/java/javaslang/collection/Comparators.java

```diff
@@ -47,7 +47,11 @@ private Comparators() {
         long serialVersionUID = 1L;
 
         static <T> SerializableComparator<T> of(Comparator<T> comparator) {
-            return comparator::compare;
+            if (comparator instanceof SerializableComparator) {
```

Comment:
Ok. We hide the constructors by making them private or package private (if needed).